src/ApiService/ApiService/onefuzzlib/Containers.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Threading.Tasks;
+using System.Linq;
+using Azure.ResourceManager;
+using Azure.Storage.Blobs;
+using Azure.Storage;
+using Azure;
+
+namespace Microsoft.OneFuzz.Service;
+
+public interface IContainers {
+    public Task<IEnumerable<byte>?> GetBlob(Container container, string name, StorageType storageType);
+
+    public BlobContainerClient? FindContainer(Container container, StorageType storageType);
+}
+
+public class Containers : IContainers {
+    private LogTracer _loggerTracer;
+    private IStorage _storage;
+    private ICreds _creds;
+    private ArmClient _armClient;
+    public Containers(ILogTracerFactory loggerFactory, IStorage storage, ICreds creds)
+    {
+        _loggerTracer = loggerFactory.MakeLogTracer(Guid.NewGuid());
+        _storage = storage;
+        _creds = creds;
+        _armClient = new ArmClient(credential: _creds.GetIdentity(), defaultSubscriptionId: _creds.GetSubcription());
```

Comment:
I think we might be ok by having ArmClient to be singleton
It's supposed to renew auth behind the scenes automatically

Reply:
we can do this in some other PR